inc/az_ulib_registry_api.h

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @file
+ *
+ * @brief   The Registry Editor.
+ *
+ * The Registry Editor is capable of storing key value pairs into device flash.
+ */
+
+#ifndef AZ_ULIB_REGISTRY_API_H
+#define AZ_ULIB_REGISTRY_API_H
+
+#include "az_ulib_pal_api.h"
+#include "az_ulib_result.h"
+#include "azure/az_core.h"
+
+#ifdef __cplusplus
+#include <cstddef>
+#include <cstdint>
+#else
+#include <stddef.h>
+#include <stdint.h>
+#endif
+
+#include "azure/core/_az_cfg_prefix.h"
+
+/**
+ * @brief   Registry control block.
+ *
+ *  Internal structure to control the registry.
+ */
+typedef struct
+{
+  /** Pointer to the start of the memory to store the registry. */
+  void* registry_start;
+
+  /** Pointer to the end of the memory to store the registry. */
+  void* registry_end;
+
+  /** Pointer to the start of the memory to store the registry information. */
+  void* registry_info_start;
+
+  /** Pointer to the end of the memory to store the registry information. */
+  void* registry_info_end;
+
+  /** Size of each page. */
+  size_t page_size;
+
+} az_ulib_registry_control_block;
+
+/**
+ * @brief   Registry information.
+ *
+ *  Structure to return registry information in the az_ulib_registry_get_info().
+ */
+typedef struct
+{
+  /** Total number of registry info that can be stored in the Registry. In number of registries. */
```

Comment:
what do you mean by "In number of registries."?

Reply:
It counts the total number of available spots to store Registry metadata.

Reply:
I think it would be better if we stick with calling each entry "node"

Reply:
Node is an implementation detail, the user of this API don't have any information about it. Any other name?

Reply:
Can you say instead: "Total number of entries available in the Registry." ?

Reply:
#Pending